src/AutoMapper/XpressionMapper/XpressionMapperVisitor.cs

```diff
@@ -231,7 +231,7 @@ protected void FindDestinationFullName(Type typeSource, Type typeDestination, st
             if (sourceFullName.IndexOf(period, StringComparison.OrdinalIgnoreCase) < 0)
             {
                 var propertyMap = typeMap.GetPropertyMaps().SingleOrDefault(item => item.DestinationProperty.Name == sourceFullName);
-                var sourceMemberInfo = typeSource.GetMember(propertyMap.DestinationProperty.Name).First();
+                var sourceMemberInfo = typeSource.GetAllMembers().First(mi => mi.Name == propertyMap.DestinationProperty.Name);

```

Comment:
There appears to be a conflict in XpressionMapperVisitor  between Type.GetMember and TypeExtensions.GetMember.  Consider calling TypeExtensions explicitly i.e. `var sourceMemberInfo = TypeExtensions.GetMember(typeSource, propertyMap.DestinationProperty.Name).First();` instead of changing the logic.